crates/utils/src/settings/mod.rs

```diff
@@ -81,15 +73,30 @@ impl Settings {
 
   pub fn save_config_file(data: &str) -> Result<String, LemmyError> {
     fs::write(CONFIG_FILE, data)?;
+    Ok(Self::read_config_file()?)
+  }
 
-    // Reload the new settings
-    // From https://stackoverflow.com/questions/29654927/how-do-i-assign-a-string-to-a-mutable-static-variable/47181804#47181804
-    let mut new_settings = SETTINGS.write().expect("write config");
-    *new_settings = match Settings::init() {
-      Ok(c) => c,
-      Err(e) => panic!("{}", e),
-    };
```

Comment:
I'm not 100% sure this is correct, since it needs also to replace itself somehow now.

Reply:
This is then only caveat to this PR then. I really couldn't find a way to hot-reload the settings, saving a new settings requires a server restart. This is because the context is behind an Arc reference, and there's no way to borrow it mutably: https://github.com/LemmyNet/lemmy/blob/remove_settings_and_secret_singletons/crates/api/src/site.rs#L483

Reply:
Damn thats true. Best I can think of is to keep accessing the settings in context, but still storing it as lazy_static. Alternatively we can merge the jwt/secrets stuff now, and keep the PR open for the part of moving settings into context, in case we have a better idea in the future.

Reply:
Okay this is done now. Still fairly clean in the way its accessed.